ibis/pyspark/compiler.py

```diff
@@ -1011,3 +1058,309 @@ def compile_cumulative_max(t, expr, scope, *, window, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.month, **kwargs)
+
+
+@compiles(ops.ExtractDay)
+def compile_extract_day(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.dayofmonth, **kwargs)
+
+
+@compiles(ops.ExtractHour)
+def compile_extract_hour(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.hour, **kwargs)
+
+
+@compiles(ops.ExtractMinute)
+def compile_extract_minute(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.minute, **kwargs)
+
+
+@compiles(ops.ExtractSecond)
+def compile_extract_second(t, expr, scope, **kwargs):
+    return _extract_x_from_datetime(t, expr, scope, F.second, **kwargs)
+
+
+@compiles(ops.ExtractMillisecond)
+def compile_extract_millisecond(t, expr, scope, **kwargs):
+    raise com.UnsupportedOperationError(
+        'PySpark backend does not support extracting milliseconds.'
+    )
+
+
+@compiles(ops.DateTruncate)
+def compile_date_truncate(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    try:
+        unit = _time_unit_mapping[op.unit]
+    except KeyError:
+        raise com.UnsupportedOperationError(
+            '{!r} unit is not supported in timestamp truncate'.format(op.unit)
+        )
+
+    src_column = t.translate(op.arg, scope)
+    return F.date_trunc(unit, src_column)
+
+
+@compiles(ops.TimestampTruncate)
+def compile_timestamp_truncate(t, expr, scope, **kwargs):
+    return compile_date_truncate(t, expr, scope, **kwargs)
+
+
+@compiles(ops.Strftime)
+def compile_strftime(t, expr, scope, **kwargs):
+    raise com.UnsupportedOperationError(
+        'PySpark uses Java SimpleDateFormat and does not support strftime.'
+    )
+
+
+@compiles(ops.TimestampFromUNIX)
+def compile_timestamp_from_unix(t, expr, scope, **kwargs):
+    op = expr.op()
+    unixtime = t.translate(op.arg, scope)
+    if not op.unit:
+        return F.to_timestamp(F.from_unixtime(unixtime))
+    elif op.unit == 's':
+        fmt = 'yyyy-MM-dd HH:mm:ss'
+        return F.to_timestamp(F.from_unixtime(unixtime, fmt), fmt)
+    else:
+        raise com.UnsupportedArgumentError(
+            'PySpark backend does not support timestamp from unix time with '
+            'unit {}. Supported unit is s.'.format(op.unit)
+        )
+
+
+@compiles(ops.TimestampNow)
+def compile_timestamp_now(t, expr, scope, **kwargs):
+    return F.current_timestamp()
+
+
+@compiles(ops.StringToTimestamp)
+def compile_string_to_timestamp(t, expr, scope, **kwargs):
+    op = expr.op()
+
+    src_column = t.translate(op.arg, scope)
+    fmt = op.format_str.op().value
+
+    if op.timezone is not None and op.timezone.op().value != "UTC":
+        raise com.UnsupportedArgumentError(
+            'PySpark backend only supports timezone UTC for converting string '
+            'to timestamp.'
+        )
+
+    return F.to_timestamp(src_column, fmt)
+
+
+@compiles(ops.DayOfWeekIndex)
+def compile_day_of_week_index(t, expr, scope, **kwargs):
```

Comment:
Maybe we can do sth similar to this?

https://github.com/databricks/koalas/blob/master/databricks/koalas/datetimes.py#L134

Reply:
So you would prefer to wrap in a Pandas UDF?

Reply:
I prefer we defer to the pandas implementation of these if possible. I am not a fan of keeping these dict of index/name mapping here.

Reply:
That makes sense. Implemented using pandas dt functions.